lib/commanded/event/handler.ex

```diff
@@ -748,31 +792,71 @@ defmodule Commanded.Event.Handler do
 
   # Delegate event to handler module.
   defp handle_event(%RecordedEvent{} = event, %Handler{} = state, context) do
-    case delegate_event_to_handler(event, state) do
+    start = System.monotonic_time()
+
+    :telemetry.execute(
+      [:commanded, :event, :handle],
+      %{system_time: System.system_time()},
+      %{recorded_event: event, handler_state: state, context: context}
+    )
+
+    handled_event = delegate_event_to_handler(event, state)
+    measurements = %{duration: System.monotonic_time() - start}
+    metadata = %{recorded_event: event, handler_state: state, context: context}
+
+    :telemetry.execute([:commanded, :event, :handle, :stop], measurements, metadata)
+
+    case handled_event do
       :ok ->
+        :telemetry.execute([:commanded, :event, :handle, :stop], measurements, metadata)
         confirm_receipt(event, state)
 
       {:ok, handler_state} ->
+        :telemetry.execute([:commanded, :event, :handle, :stop], measurements, metadata)
```

Comment:
The updated `handler_state` should be included in the telemetry `metadata`.

Reply:
@slashdotdash updated.